src/binding/python/RecordComponent.cpp

```diff
@@ -51,20 +51,86 @@ void init_RecordComponent(py::module &m) {
+            //    https://github.com/pybind/pybind11/issues/1224#issuecomment-354357392
+            // scalars, see PEP 3118
+            // requires Numpy 1.15+
+            if( buf.ndim == 0 ) {
+                // refs:
+                //   https://docs.scipy.org/doc/numpy-1.15.0/reference/arrays.interface.html
+                //   https://docs.python.org/3/library/struct.html#format-characters
+                // std::cout << "  scalar type '" << buf.format << "'" << std::endl;
+                // typestring: encoding + type + number of bytes
+                switch( dtype )
+                {
+                    case DT::BOOL:
+                        return rc.makeConstant( *static_cast<bool*>(buf.ptr) );
+                        break;
+                    case DT::SHORT:
+                        return rc.makeConstant( *static_cast<short*>(buf.ptr) );
+                        break;
+                    case DT::INT:
+                        return rc.makeConstant( *static_cast<int*>(buf.ptr) );
+                        break;
+                    case DT::LONG:
+                        return rc.makeConstant( *static_cast<long*>(buf.ptr) );
+                        break;
+                    case DT::LONGLONG:
+                        return rc.makeConstant( *static_cast<long long*>(buf.ptr) );
+                        break;
+                    case DT::USHORT:
+                        return rc.makeConstant( *static_cast<unsigned short*>(buf.ptr) );
+                        break;
+                    case DT::UINT:
+                        return rc.makeConstant( *static_cast<unsigned int*>(buf.ptr) );
+                        break;
+                    case DT::ULONG:
+                        return rc.makeConstant( *static_cast<unsigned long*>(buf.ptr) );
+                        break;
+                    case DT::ULONGLONG:
+                        return rc.makeConstant( *static_cast<unsigned long long*>(buf.ptr) );
+                        break;
+                    case DT::FLOAT:
+                        return rc.makeConstant( *static_cast<float*>(buf.ptr) );
+                        break;
+                    case DT::DOUBLE:
+                        return rc.makeConstant( *static_cast<double*>(buf.ptr) );
+                        break;
+                    case DT::LONG_DOUBLE:
+                        return rc.makeConstant( *static_cast<long double*>(buf.ptr) );
+                        break;
+                    default:
+                        throw std::runtime_error("make_constant: "
+                            "Unknown Datatype!");
+                }
+            }
+            else
+            {
+                throw std::runtime_error("make_constant: "
+                    "Only scalar values supported!");
+            }
+
+        }, py::arg("value")
+        )
+        // allowed python intrinsics, after (!) buffer matching
+        .def("make_constant", &RecordComponent::makeConstant<char>,
```

Comment:
must be `std::string`: Python has no chars and otherwise our strings are all casted to size-one char

Reply:
but a `std::string` makes no real sense as a value for a record. so let's keep `char`s here.